cli/src/main/java/com/devonfw/tools/ide/network/NetworkStatusImpl.java

```diff
@@ -0,0 +1,135 @@
+ */
+public class NetworkStatusImpl implements NetworkStatus {
+
+  private final AbstractIdeContext context;
+
+  private NetworkProxy networkProxy;
+
+  private final String onlineCheckUrl;
+
+  protected final CachedValue<Throwable> onlineCheck;
+
+  /**
+   * @param ideContext the {@link AbstractIdeContext}.
+   */
+  public NetworkStatusImpl(AbstractIdeContext ideContext) {
+    this(ideContext, null, CachedValue.DEFAULT_RETENTION);
+  }
+
+  /**
+   * @param context the {@link AbstractIdeContext}.
+   * @param onlineCheckUrl the URL to test for the online-check.
+   * @param retention the retention of the {@link CachedValue}.
+   */
+  protected NetworkStatusImpl(AbstractIdeContext context, String onlineCheckUrl, long retention) {
+    this.context = context;
+    if (onlineCheckUrl == null) {
+      onlineCheckUrl = "https://www.github.com";
+    }
+    this.onlineCheckUrl = onlineCheckUrl;
+    this.onlineCheck = new CachedValue<>(this::doOnlineCheck, retention);
+  }
+
+  @Override
+  public boolean isOfflineMode() {
+
+    return this.context.isOfflineMode();
+  }
+
+  @Override
+  public boolean isOnline() {
+
+    return getError() == null;
+  }
+
+  @Override
+  public Throwable getError() {
+
+    return this.onlineCheck.get();
+  }
+
+  private Throwable doOnlineCheck() {
+    configureNetworkProxy();
+    try {
+      int timeout = 1000;
+      //open a connection to URL and try to retrieve data
+      //getContent fails if there is no connection
+      URLConnection connection = new URL(this.onlineCheckUrl).openConnection();
+      connection.setConnectTimeout(timeout);
+      connection.getContent();
+      return null;
+    } catch (Exception e) {
+      if (this.context.debug().isEnabled()) {
+        this.context.debug().log(e, "Error when trying to connect to {}", this.onlineCheckUrl);
+      }
+      return e;
+    }
+  }
+
+  private void configureNetworkProxy() {
+
+    if (this.networkProxy == null) {
+      this.networkProxy = new NetworkProxy(this.context);
+      this.networkProxy.configure();
+    }
+  }
+
+  @Override
+  public void logStatusMessage() {
+
+    if (isOfflineMode()) {
+      this.context.warning("You are offline because you have enabled offline mode via CLI option.");
+      return;
+    }
+    Throwable error = getError();
+    if (error == null) {
+      this.context.success("You are online.");
+      return;
+    }
+    String message = "You are offline because of the following error:";
+    if (this.context.debug().isEnabled()) {
+      this.context.error(error, message);
+    } else {
+      this.context.error(message);
+      this.context.error(error.toString());
+    }
+    if (error instanceof SSLException) {
+      this.context.warning(
+          "You are having TLS issues. We guess you are forced to use a VPN tool breaking end-to-end encryption causing this effect. As a workaround you can call the following command:");
+      this.context.interaction("ide fix-vpn-tls-problem");
```

Comment:
Untill we have implemented the suggestion in #897 and have the `fix-vpn-tls-problem` commandlet, we should log something else here:

```suggestion
          "You are having TLS issues. We guess you are forced to use a VPN tool breaking end-to-end encryption causing this effect. As a workaround you can create and configure a truststore as described here:");
      this.context.interaction("https://github.com/devonfw/IDEasy/blob/main/documentation/proxy-support.adoc#tls-certificate-issues");
```